### ISSUE TITLE
feat(language): build transformer block with layer norm

### ISSUE THREAD

**Issue by leandronsp** [leandronsp/aspirina#5]
## Overview

Implement the transformer block in `language/transformer.rs`: multi-head attention + feed-forward network + layer normalization + residual connections. Add GELU activation to `core/calc.rs`. This is the stackable unit. N of these blocks make the model.

## Problem Statement

Attention alone isn't enough. The transformer block wraps attention with normalization (keeps values stable across layers), a feed-forward network (adds non-linear capacity), and residual connections (lets gradients flow through deep stacks). Without these, stacking more than 2-3 layers fails.

## Current State

- Multi-head attention exists in `language/attention.rs` (from issue #4)
- `core/calc.rs` has sigmoid, tanh, softmax. No GELU, no layer norm
- `core/matrix.rs` handles all matrix operations

## Requirements

- [ ] R-1: Add GELU activation function to `core/calc.rs` with derivative
- [ ] R-2: Implement layer normalization: normalize across the feature dimension, with learnable scale and bias parameters
- [ ] R-3: Implement feed-forward network (FFN): two linear layers with GELU activation between them. Hidden dim typically 4x embed_dim
- [ ] R-4: Implement transformer block: LayerNorm -> MultiHeadAttention -> residual add -> LayerNorm -> FFN -> residual add (Pre-LN architecture)
- [ ] R-5: Make blocks stackable: `Vec<TransformerBlock>` where output of block N feeds into block N+1
- [ ] R-6: Forward and backward pass through the full stack
- [ ] R-7: Unit tests for layer norm, GELU, FFN, single block, stacked blocks

## Technical Approach

```
language/src/
  transformer.rs  # LayerNorm, FFN, TransformerBlock structs

core/src/
  calc.rs         # add gelu(), gelu_derivative()
```

Pre-LN transformer block (more stable for training):
```
x = input
x = x + multi_head_attention(layer_norm(x))    # residual + attention
x = x + ffn(layer_norm(x))                     # residual + FFN
```

Start with 2 blocks. The live demonstrates that adding more blocks = more capacity.

## Acceptance Criteria

- Given input (4, 64) through 2 stacked transformer blocks, when I run forward pass, then output is (4, 64) with different values
- Given layer_norm on a vector, when I compute it, then mean is ~0 and variance is ~1
- Given GELU(0), when I compute it, then the result is 0.0
- Given residual connections, when I compare input and output of a block, then the output is input + transformation (not just transformation)

## Out of Scope

- Positional encoding (can be added to embedding or as separate module)
- Dropout regularization
- KV-cache for inference optimization
